kontonummer.js

```diff
@@ -20,406 +20,406 @@
+        modulo  : 11,
+        lengths : {
+            clearing : 4,
+            account  : 7,
+            control  : 10
+        }
+    },{
+        name    : 'Skandiabanken',
+        regex   : /^(91[5-6][0-9])([0-9]{7})$/,
+        modulo  : 11,
+        lengths : {
+            clearing : 4,
+            account  : 7,
+            control  : 11
+        }
+    },{
+        name    : 'Sparbanken Syd',
+        regex   : /^(957[0-9])([0-9]{10})$/,
+        modulo  : 10,
+        lengths : {
+            clearing : 4,
+            account  : 10,
+            control  : 10
+        }
+    },{
+        name    : 'Swedbank',
+        regex   : /^(7[0-9][0-9][0-9])([0-9]{7})$/,
+        modulo  : 11,
+        lengths : {
+            clearing : 4,
+            account  : 7,
+            control  : 10
+        }
+    },{
+        name    : 'Swedbank',
+        regex   : /^(93[0-2][0-9])([0-9]{10})$/,
+        modulo  : 10,
+        lengths : {
+            clearing : 4,
+            account  : 10,
+            control  : 10
+        }
+    },{
+        name    : 'Swedbank',
+        regex   : /^(8[0-9]{4})([0-9]{10})$/,
+        modulo  : 10,
+        lengths : {
+            clearing : 5,
+            account  : 10,
+            control  : 10
+        }
+    },{
+        name    : 'Ålandsbanken',
+        regex   : /^(23[0-9][0-9])([0-9]{7})$/,
+        modulo  : 11,
+        lengths : {
+            clearing : 4,
+            account  : 7,
+            control  : 11
+        }
+    }];
+
+    for (i in banks) {
+        b = banks[i];
+        cn = n.substr(-b.lengths.control, b.lengths.control);
+        if (b.regex.test(n) && ((b.modulo === 11 && mod11(cn)) || (b.modulo === 10 && mod10(cn)))) {
+            return {
+                bank_name       : b.name,
+                clearing_number : n.substr(0, b.lengths.clearing),
+                account_number  : n.substr(b.lengths.clearing, b.lengths.account)
+            };
+        }
+    }
+    return false;
+};
+
+/**
+ * Stödfunktion för att kontrollera mod10.
+ * 
+ * @param {String} number Bankkontonummer
+ * @returns {Boolean}
+ */
+var mod10 = function (number) {
+    var len = number.length, bit = 1, sum = 0, val, arr = [0, 2, 4, 6, 8, 1, 3, 5, 7, 9];
+    while (len) {
+        val = parseInt(number.charAt(--len), 10);
+        sum += (bit ^= 1) ? arr[val] : val;
+    }
+    return sum && sum % 10 === 0;
+};
+
+/**
+ * Stödfunktion för att kontrollera mod11.
+ * 
+ * @param {String} number Bankkontonummer
+ * @returns {Boolean}
+ */
+var mod11 = function (number) {
+    var len = number.length, sum = 0, val, weights = [1, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1];
+    var arr = weights.splice(weights.length-len, weights.length-(weights.length-len));
+    while (len) {
+        val = parseInt(number.charAt(--len), 10);
+        sum += arr[len] * val;
+    }
+    return sum && sum % 11 === 0;
+};
+
+module.exports = kontonummer;
```

Comment:
This approach, while works with node, makes the code unusable if just included as a script on a page, since `module` is undefined. Apart from that, it would add the `mod11` and `mod10` functions to the global namespace. I would suggest to keep the IIFE but add such a claue:
```js

    if (typeof module !== 'undefined' && module.exports) {
      module.exports = kontonummer;
    } else if (typeof window !== 'undefined') {
      window.kontonummer = kontonummer;
    }
``` 

This keeps it backwards compatible